### PR TITLE
Remove redundant code for EOL Python <= 3.6

### PR DESCRIPTION
**Pull request by hugovk**
### Description

There's a few bits and pieces of code to support Python <= 3.6 which are no longer needed and can be removed, to slightly simplify the codebase.

### Checklist

This pull request is:

- [ ] A documentation / typographical error fix
	- Good to go, no issue or tests are needed
- [x] A short code fix
	- please include the issue number, and create an issue if none exists, which
	  must include a complete example of the issue.  one line code fixes without an
	  issue and demonstration will not be accepted.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.   one line code fixes without tests will not be accepted.
- [ ] A new feature implementation
	- please include the issue number, and create an issue if none exists, which must
	  include a complete example of how the feature would look.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.

**Have a nice day!**

### CODE REVIEW

File: test/requirements.py [hunk 1]
```diff
@@ -1393,26 +1391,6 @@ def has_fastexecutemany(config):
...
 
-    @property
-    def python_fixed_issue_8743(self):
-        return exclusions.skip_if(
-            lambda: sys.version_info < (2, 7, 8),
-            "Python issue 8743 fixed in Python 2.7.8",
-        )
-
-    @property
-    def granular_timezone(self):
```
Comment (zzzeek) [inline test/requirements.py:1404]: it appears this req was not being used anyway.

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **zzzeek** to try to get revision 282b4a91282902a57807aa2541b75b272b547127 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
New Gerrit review created for change 282b4a91282902a57807aa2541b75b272b547127: https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/3492

**Comment by sqla-tester** [comment]
Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/3492 has been **merged**. Congratulations! :)

**Pull request closed without merging**